src/main/java/strategy/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Storage storage;
```

Comment:
According to the SOLID principles and the checklist, the `Storage` class should not be instantiated inside the `ShopServiceImpl` class. Instead, it should be used as a dependency and passed to the service if necessary. This will make the service more independent and easier to test.